source/backend/tests/unit/dal/Repositories/DispositionFilePropertyRepositoryTest.cs

```diff
@@ -104,6 +104,126 @@ public void GetDispositionFilePropertyRelatedCount_NotFound()
         }
         #endregion
 
+        #region Add
+        [Fact]
+        public void Add_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.DispositionAdd);
+
+            var context = helper.CreatePimsContext(user, true);
+            var repository = helper.CreateRepository<DispositionFilePropertyRepository>(user);
+
+            var pimsDispositionFileProperty = new PimsDispositionFileProperty() { DispositionFileId = 1, PropertyId = 1, Property = new PimsProperty() { RegionCode = 1, PropertyId = 1 } };
+
+            // Act
+            var result = repository.Add(pimsDispositionFileProperty);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeAssignableTo<PimsDispositionFileProperty>();
+        }
+
+        [Fact]
+        public void Add_ThrowIfNull()
+        {
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.DispositionAdd);
+            helper.CreatePimsContext(user, true);
+
+            var repository = helper.CreateRepository<DispositionFilePropertyRepository>(user);
+
+            // Act
+            Action act = () => repository.Add(null);
+
+            // Assert
+            act.Should().Throw<ArgumentNullException>();
+        }
+        #endregion
+
+        #region Update
+        [Fact]
+        public void Update_Success()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.DispositionAdd);
```

Comment:
nit: Repositories don't check for permissions but for consistency it should be DispositionUpdate here as we are testing the update method